src/Document/Dictionary/DictionaryValue/Date/DateValue.php

```diff
@@ -15,32 +15,54 @@ public function __construct(
     ) {
     }
 
+    /***
+     * @param string $valueString
+     * @return self|null
+     * @throws InvalidArgumentException non-hex characters detected.
+     */
     #[Override]
     public static function fromValue(string $valueString): ?self {
-        if (str_starts_with($valueString, '<') && str_ends_with($valueString, '>')) {
-            $valueString = substr($valueString, 1, -1);
-            if (!ctype_xdigit($valueString) || strlen($valueString) % 2 !== 0) {
-                throw new InvalidArgumentException(sprintf('String "%s" is not hexadecimal', substr($valueString, 0, 10)));
-            }
-
-            $valueString = hex2bin($valueString);
-            if ($valueString === false) {
+        // FEFF0044003A00320030003200300030003900330030003100350034003300300037005A
+        // UTF-16BE encoded string.
+        if (preg_match('/<FEFF([A-Z0-9]{68})>/', $valueString, $matches) !== false) {
```

Comment:
I might be wrong here, but shouldn't you test if `preg_match` returns `1`? Unequal to false only means, there was no error, but not if the pattern matches.